### ISSUE TITLE
Suggestion: Add filter for "Account Build" quests

### ISSUE THREAD

**Issue by Simpuhl** [Zoinkwiz/quest-helper#1668]
I have a zerker pure and a couple of other accounts and I get a lot of questions in game about what quests to do etc for these account types. When i look online, many of the guides seem outdated. I was thinking maybe Quest Helper could add these under the Filters? 

Basically add "Account Build" quests for example:

1 Def Pure Quests
Berserker Quests
Obsidian mauler Quests
Etc.

![image 1](https://github.com/user-attachments/assets/fe31243a-4404-47bd-abdf-200ab3ced601)

**Comment by Zoinkwiz** [comment]
Hey there, is the existing skill filtering useful for these account types, or are there additional considerations outside of skill requirements and experience rewards?

**Comment by Simpuhl** [comment]
Specific Account type will need specific quests, example a zerker needs to do certain quests to get barrows gloves and they need to not accept certain xp rewards and such. Some are easier than others.

Another way to look at this might be 

Berserker Friendly Quests - meaning all quests a zerker can do to not hurt the account build
Berserker Required Quests - Minimum Quests needed for the build and specific order

The wiki lists alot of these requirements already for example - https://oldschool.runescape.wiki/w/Berserker_(Account_build)#NH/Tribrid

I wouldn't say all of them would be needed but definitely would be a nice little upgrade to the tool and help a lot of people, currently they google search and come across some sketchy websites.

**Comment by EthonSmoth** [comment]
Coming over from the Discussion thread, Zoinkwiz pointed me here, and this is the right place for it. Caught up on the thread and on the wiki pages Simpuhl linked.

To answer the question about whether the existing skill filter is enough: I don't think it is, and Simpuhl's friendly/required split is the reason. Skill filtering handles "can the player do this quest", but not "will doing this quest break the build." The wiki pages make the gap pretty explicit — the Obsidian mauler page says outright that obbies are "Quest-locked out of any quest rewards that grant Attack experience," citing Waterfall Quest, the Gnome series (including Monkey Madness I and spirit tree access), and Death Plateau as off-limits. None of that shows up in a skill filter, because the player meets the skill requirements fine — it's the reward that breaks the build.

A few patterns emerge from the wiki pages that I think are worth designing around:

XP rewards that overshoot caps. The Berserker page lists 10 quests that, done in the right order, leave the account at 44 Defence with 4,875 XP to 45 (i.e. the player gets to choose when to cross the threshold). Skill filter can't represent that.

Branching sub-builds. "Obsidian mauler" isn't one preset; the page lays out conditional quest paths for 1 Def, 16 Att, 13 Def, 20 Def, 30 Def, and 40+ Def variants, each with different quest sets. A preset system would need to handle build families, not just builds.

Quirks the wiki documents that skill logic misses. The obby guide notes you don't actually need 20 Attack for Merlin's Crystal because you can right-click Excalibur. Quest Helper already handles things like this well; embedded presets could surface them in context (e.g. "this quest looks blocked but is actually doable on this build because…").

So a build preset is realistically:

- A set of stat caps (partly covered by existing skill logic)
- Required quests in a specific order, with notes on reward choices (e.g. Monkey Madness I — "ENSURE you take the Strength and Hitpoints experience option")
- Friendly quests (safe to do whenever)
- Breaking quests (warn or hide)
- Optionally: conditional branches for sub-builds

The good news is that the wiki maintains all of this already. The Berserker page has a complete 10-quest table with reward notes; the Obsidian mauler page has the mandatory list plus the branching variants. So the work for phase 1 is mostly transcription, not original research.

Given the steer in the Discussion thread (embedded presets first, JSON import later), this maps cleanly to Simpuhl's framing. A couple of questions before drafting anything:

Would it help if I put together a candidate list of builds + their classified quest lists (sourced from the wiki) as a markdown table on this issue for review, before any code lands? Lower-risk way to sanity-check the data than putting it straight into a PR. I'd start with Berserker and 1 Def Obby Pure since both have well-maintained wiki pages.

For the sub-build problem (e.g. Obby Mauler having 6+ variants): would the team prefer one preset per variant, or one preset per build family with a sub-selector? The former is simpler to ship; the latter handles the wiki's actual structure more faithfully.

On reward choices like Monkey Madness I's XP option, is there existing infrastructure for surfacing "take option X" guidance in the helper sidebar, or would that be new?

Thanks for the steer,  much better scope than what I originally pitched.

**Comment by pajlada** [comment]
Put together a list of account builds (including variants) users would want to have a custom order/filter for, and figure out which are easy to implement (i.e. they _only_ need a specific set of quests, no half-finished quests), and which require extra messaging (i.e. complete these quests, but ONLY finish this quest up to this point)

**Comment by EthonSmoth** [comment]
Put together an initial list of candidate builds grouped by implementation complexity.

"Easy" here means the build can mostly be represented as a list of quests to complete or avoid. "Harder" means the build depends on extra messaging such as reward choices, XP lamps, or intentionally stopping part-way through a quest.

## Tier 1 — Straightforward quest lists

These should be expressible as curated quest lists without needing much extra logic.

| Build | Why it is straightforward | Notes |
| --- | --- | --- |
| Main / Quest Cape route | No account restrictions; just ordering | Useful baseline preset |
| Ironman efficient route | Mostly ordering/routing | Could use existing wiki routing as source |
| HCIM cautious route | Mostly ordering with risk-aware sequencing | Similar to Ironman, but safer ordering |
| F2P main | Quest subset only | Simple members/non-members split |
| F2P skiller | Quest subset + combat XP exclusions | Still fairly finite |
| Combat-only pure | Mostly exclusion-based | Small quest pool |

## Tier 2 — Needs reward-choice messaging

These builds can still be represented as quest lists, but some quests need prominent warnings or notes.

| Build | Issue | Example messaging needed |
| --- | --- | --- |
| Berserker | Defence/Attack/Prayer reward routing matters | "Take this reward option, not that one" |
| 1 Defence Pure | Must avoid Defence XP quests and subquests | "Do not complete this subquest if preserving 1 Defence" |
| Obsidian mauler variants | Attack XP rewards can break the build | "This quest gives Attack XP and is not safe for this variant" |
| Void pure / moon pure style builds | Defence thresholds matter | "Only safe if this Defence target is intended" |

This probably only needs a lightweight `note`, `warning`, or `reward_note` field attached to specific quest entries. Example:

```json
{
  "quest": "monkey_madness_i",
  "status": "required",
  "note": "Ensure you take the Strength and Hitpoints experience option."
}
```

## Tier 3 — Needs partial-completion or "do not finish" messaging

These are harder because the correct route may involve starting a quest, getting an unlock, and then intentionally not completing it.

| Build | Quest / situation | Why it is harder |
| --- | --- | --- |
| 1 Defence Obby Pure | Dragon Slayer I | May want access/unlocks but completion gives Defence XP |
| Obby / low-Attack variants | Mountain Daughter and similar quests | Quest progress/rewards can matter before completion |
| Berserker | Monkey Madness I / Daero reward handling | Completion and reward claiming may need separate guidance |
| Defence-capped builds | XP lamps / deferred rewards | The player may need to hold or route rewards carefully |
| Skillers | Partial-access quests | Some quests may be useful only up to a safe stopping point |

This is probably the main design question: **should "intentionally incomplete" be a first-class state, or is a warning/note enough?**

Possible structure if it needs to be first-class:

```json
{
  "quest": "dragon_slayer_i",
  "status": "partial",
  "stop_at": "after_obtaining_required_unlock",
  "warning": "Do not complete the quest if preserving this build, as completion grants Defence XP."
}
```

## Tier 4 — Build families / variants

Some builds are not single presets but families with variants. These could either be implemented as separate presets or as one build family with a sub-selector.

| Build family | Possible variants |
| --- | --- |
| Obsidian mauler | 1 Defence, 13 Defence, 16 Attack, 20 Defence, 30 Defence, 40+ Defence, Obby Zerker |
| Low Defence pure | 1 Defence, 13 Defence, 20 Defence, 30 Defence |
| F2P pure | 1 Defence pure, Strength pure, Rune pure, Adamant pure, Tank, Magic tank |
| Range tank | Multiple Defence thresholds |

## Suggested phase 1 scope

For a first pass, I'd suggest avoiding partial-completion builds unless the team wants to model that explicitly from the start.

A practical phase 1 could be:

- Main / Quest Cape route
- Ironman efficient route
- Skiller or F2P skiller
- Berserker
- 1 Defence Pure

That would test normal ordering, exclusion-based filtering, XP reward warnings, and build-breaking quest warnings. Phase 2 could then handle intentionally incomplete quests, build-family sub-selectors, user-imported JSON lists, and more niche variants.

## Open questions

1. Should "intentionally incomplete" quests be modelled as their own state, or handled as warnings/notes?
2. For build families like Obsidian mauler, would maintainers prefer one preset per variant or one preset with a variant selector?
3. Are there particular builds the team sees more demand for than the ones listed above?

Happy to double-check any specific build against the wiki before committing it to a preset — I've kept exact quest XP numbers out for now since those go stale, and the wiki's Quest Experience Rewards page is the better source of truth for the plugin to link to anyway.

**Comment by Simpuhl** [comment]
Yeah, id start with a few of the most popular ones and then add more later
